core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -165,8 +165,13 @@ public void initialize(String name, Map<String, String> unresolved) {
             CatalogProperties.AUTH_DEFAULT_REFRESH_ENABLED_DEFAULT);
     this.client = clientBuilder.apply(mergedProps);
     this.paths = ResourcePaths.forCatalogProperties(mergedProps);
+    int tokenRefreshRetries =
+        PropertyUtil.propertyAsInt(
+            props,
+            OAuth2Properties.TOKEN_REFRESH_NUM_RETRIES,
+            OAuth2Properties.TOKEN_REFRESH_NUM_RETRIES_DEFAULT);
 
-    this.catalogAuth = new AuthSession(baseHeaders, null, null, credential);
+    this.catalogAuth = new AuthSession(baseHeaders, null, null, credential, tokenRefreshRetries);
```

Comment:
I'm not a fan of adding more properties to `AuthSession`. Maybe we should have a config object that we pass around with the default refresh interval, retry settings (including the others), and the OAuth scope?

Reply:
We discussed this offline and concluded that we don't want to expose `OAuth2Properties.TOKEN_REFRESH_NUM_RETRIES` and make it externally configurable. 
We agreed to add a `public static` field to `AuthSession` annotated with `@VisibleForTesting`.